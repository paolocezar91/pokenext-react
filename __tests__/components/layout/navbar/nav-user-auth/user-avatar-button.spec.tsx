import UserAvatarButton from "@/components/layout/navbar/nav-user-auth/user-avatar-button";
import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";

describe("UserAvatarButton", () => {
  const clickCallback = jest.fn();
  const onMouseEnterCallback = jest.fn();

  it("renders correctly with image", () => {
    render(
      <UserAvatarButton
        image={"image.png"}
        onClick={() => clickCallback()}
        onMouseEnter={() => onMouseEnterCallback()}
      ></UserAvatarButton>,
    );

    const img = document.querySelector("img");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "image.png");
  });

  it("tests clicks and onMouseEnter without image", () => {
    render(
      <UserAvatarButton
        image="image.png"
        onClick={() => clickCallback()}
        onMouseEnter={() => onMouseEnterCallback()}
      ></UserAvatarButton>,
    );

    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(clickCallback).toHaveBeenCalled();
    fireEvent.mouseEnter(button);
    expect(onMouseEnterCallback).toHaveBeenCalled();
  });

  it("should match snapshot with image", () => {
    const { asFragment } = render(
      <UserAvatarButton
        image="image.png"
        onClick={() => clickCallback()}
        onMouseEnter={() => onMouseEnterCallback()}
      ></UserAvatarButton>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly without image", () => {
    render(
      <UserAvatarButton
        image={undefined}
        onClick={() => clickCallback()}
        onMouseEnter={() => onMouseEnterCallback()}
      ></UserAvatarButton>,
    );

    const svg = document.querySelector("svg");
    expect(svg).toBeInTheDocument();
    expect(svg).toHaveClass("h-6", "w-6", "text-white");
  });

  it("tests clicks and onMouseEnter without image", () => {
    render(
      <UserAvatarButton
        image={undefined}
        onClick={() => clickCallback()}
        onMouseEnter={() => onMouseEnterCallback()}
      ></UserAvatarButton>,
    );

    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(clickCallback).toHaveBeenCalled();
    fireEvent.mouseEnter(button);
    expect(onMouseEnterCallback).toHaveBeenCalled();
  });

  it("should match snapshot with image", () => {
    const { asFragment } = render(
      <UserAvatarButton
        image={undefined}
        onClick={() => clickCallback()}
        onMouseEnter={() => onMouseEnterCallback()}
      ></UserAvatarButton>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
