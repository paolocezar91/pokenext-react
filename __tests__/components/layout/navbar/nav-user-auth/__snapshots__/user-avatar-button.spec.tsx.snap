// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatarButton should match snapshot with image 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
    type="button"
  >
    <img
      alt="User avatar"
      class="w-10 h-10 rounded-full border-2 border-gray-300 object-cover opacity-90 hover:opacity-100"
      src="image.png"
    />
  </button>
</DocumentFragment>
`;

exports[`UserAvatarButton should match snapshot with image 2`] = `
<DocumentFragment>
  <button
    class="
    flex
    px-2
    py-2
    text-sm
    rounded
    cursor-pointer
    transition-colors
    hover:shadow-md
    hover:bg-(--pokedex-red-dark)
    active:bg-white
    active:text-(--pokedex-red-dark)
   bg-transparent  btn btn-primary"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="h-6 w-6 text-white"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M7.5 6a4.5 4.5 0 1 1 9 0 4.5 4.5 0 0 1-9 0ZM3.751 20.105a8.25 8.25 0 0 1 16.498 0 .75.75 0 0 1-.437.695A18.683 18.683 0 0 1 12 22.5c-2.786 0-5.433-.608-7.812-1.7a.75.75 0 0 1-.437-.695Z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;
